front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx

```diff
@@ -68,29 +69,28 @@ const formatPointWithNoName = (
 const extractMarkerInformation = (pathSteps: MarkerInformation[], showStdcmAssets: boolean) =>
   pathSteps.reduce((acc, cur, index) => {
     if (cur && cur.coordinates) {
-      if (index === 0) {
+      if (cur.pointType === MARKER_TYPE.ORIGIN) {
         acc.push({
           coordinates: cur.coordinates,
           type: MARKER_TYPE.ORIGIN,
           marker: cur,
           imageSource: showStdcmAssets ? stdcmOrigin : originSVG,
         });
-      } else if (index > 0 && index < pathSteps.length - 1) {
+      } else if (cur.pointType === MARKER_TYPE.DESTINATION) {
         acc.push({
           coordinates: cur.coordinates,
-          type: MARKER_TYPE.VIA,
+          type: MARKER_TYPE.DESTINATION,
           marker: cur,
-          imageSource: showStdcmAssets ? stdcmVia : viaSVG,
-          index,
+          imageSource: showStdcmAssets ? stdcmDestination : destinationSVG,
         });
-      } else if (index === pathSteps.length - 1) {
+      } else if (cur.pointType === MARKER_TYPE.VIA)
```

Comment:
This last condition is not necessary ;)
```suggestion
      } else {
```

Reply:
Done